pkg/api/serialization_test.go

```diff
@@ -144,6 +146,40 @@ func roundTripSame(t *testing.T, group testapi.TestGroup, item runtime.Object, e
 	}
 }
 
+func TestSetControllerConversion(t *testing.T) {
+	rs := &extensions.ReplicaSet{}
+	rc := &v1.ReplicationController{}
+
+	extGroup := testapi.Extensions
+	defaultGroup := testapi.Default
+
+	fuzzInternalObject(t, extGroup.InternalGroupVersion(), rs, rand.Int63())
+
+	// RS -> RC
+	data, err := runtime.Encode(extGroup.Codec(), rs)
+	if err != nil {
+		t.Fatalf("unexpected encoding error: %v", err)
+	}
+
+	codec := extGroup.Codec()
```

Comment:
`decoder := api.Codecs.UniversalDecoder(*extGroup.GroupVersion(), *defaultGroup.GroupVersion())` will get you to the next error.  Looks like its going to to fail in a reflecting conversion path for `'v1beta1.ReplicaSet' to 'v1.ReplicationController'`.